buildpacks/nodejs-npm-install/src/npm.rs

```diff
@@ -43,12 +43,17 @@ impl<'a> From<Version<'a>> for Command {
 
 pub(crate) struct Install<'a> {
     pub(crate) env: &'a Env,
+    pub(crate) has_lockfile: bool,
 }
 
 impl<'a> From<Install<'a>> for Command {
     fn from(value: Install<'a>) -> Self {
         let mut cmd = Command::new("npm");
-        cmd.arg("ci");
+        if value.has_lockfile {
+            cmd.arg("ci");
+        } else {
+            cmd.args(["install", "--no-package-lock"]);
```

Comment:
I still don't love this. I don't think we should be supporting apps without lockfiles. We have a solution for supporting functions without lockfiles already in https://github.com/heroku/buildpacks-nodejs/pull/706.

Reply:
👍 agreed. going forward with #706